### ISSUE TITLE
Could not find a declaration file for module 'convert-units'

### ISSUE THREAD

**Issue by oba93** [convert-units/convert-units#151]
I have error inside vscode like this:

> Could not find a declaration file for module 'convert-units'

At the line:

`import * as convertUnits from 'convert-units';`

In the browser i have something like this:

> Uncaught ReferenceError: global is not defined

If i try do something like this:

`const convert = require('convert-units');`

I getting this inside console:

> ERROR in src/app/canvas/components/side-window/components/side-window-form/side-window-form.component.ts(28,26): error TS2591: Cannot find name 'require'. Do you need to install type definitions for node? Try `npm i @types/node` and then add `node` to the types field in your tsconfig.

Somebody has idea how to import this module correctly? I try npm install and other solutions found in google but nothing works.

**Comment by ararog** [comment]
Add this to dependencies section of package.json:  `"convert-units": "github:ben-ng/convert-units"`

**Comment by kayleighbray** [comment]
@oba93 Looks like you're using Angular. Try adding `(window as any).global = window;` to your polyfills file, that worked for me

**Comment by emmkimme** [comment]
the file lib/index.d.ts is not shipped in the package. It is why, typescript import does not work.
Would appreciate if a new package is generated with this file.
Thanks

**Comment by EricWiener** [comment]
I think you can just install from https://www.npmjs.com/package/@types/convert-units.

**Comment by emmkimme** [comment]
Sorry, I was not clear, the @types d.ts implied to activate a 'esModuleInterop' flag as it is not compliant with es6 modules spec.
> This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.ts
Would be better to review the api and shipped new definitions directly from package. Understand it is not a priority but just more comfortable for TS developers.

**Comment by jgadbois** [comment]
I'm also getting the `ReferenceError: global is not defined` on version 2.3.4 inside an Ember application.  I'm including as specified in the docs

```
import configureMeasurements, { allMeasures } from 'convert-units';
```

The error appears to be coming from lodash.

**Comment by Taar** [comment]
> 
> 
> I'm also getting the `ReferenceError: global is not defined` on version 2.3.4 inside an Ember application. I'm including as specified in the docs
> 
> ```
> import configureMeasurements, { allMeasures } from 'convert-units';
> ```
> 
> The error appears to be coming from lodash.

If you're using version 2.3.4, you'll need to refer to it's [documentation](https://www.npmjs.com/package/convert-units/v/2.3.4). Support for the import snippet above has not yet been released to npm.

**Comment by jgadbois** [comment]
I get the same error with

`import convert from 'convert-units';`

And 

`var convert = require('convert-units')`

**Comment by jzybert** [comment]
Any update on this? I'm getting this error currently in my React application

**Comment by Taar** [comment]
> ~Any update on this? I'm getting this error currently in my React application~
> 
> I switched to version `^3.0.0-beta.2` which has better TypeScript support and was able to import it with errors

Are you still getting errors or has version `^3.0.0-beta.2` resolved those errors?

**Comment by jzybert** [comment]
Sorry, typo! without errors is what I meant to say. I'll update the original comment

**Comment by faribauc** [comment]
Following up on this discussion, I'm getting the following error:

```
Cannot find module 'convert-units/definitions/all' or its corresponding type declarations.
```

I've installed `@types/convert-units` and running on version 2.3.4 of the `convert-units` package.

Am I missing any more types?

**Comment by Taar** [comment]
> I've installed `@types/convert-units` and running on version 2.3.4 of the `convert-units` package.

@faribauc Make sure you're looking at the documentation for the version you're using, [`2.3.4`](https://www.npmjs.com/package/convert-units). The import statements you have there will only work for the [latest beta release](https://www.npmjs.com/package/convert-units/v/3.0.0-beta.5). If you decide to use a beta version, note that `@types/convert-units` is not required for the beta versions as the project is now written in TS and bundles the definitions (at least is should, so let me know if it's not working).

**Comment by faribauc** [comment]
> 

@Taar I tried again with the beta version (and this time removed `@types/convert-units`) but it's still showing the error I mentionned.

**Comment by faribauc** [comment]
@Taar Are the definitions included in the beta version? I don't see them in the code.

**Comment by Taar** [comment]
@faribauc the current docs for the `main` branch have not been released yet. I'm hoping to get everything squared away and ready to release soon. I recommend looking at the docs for [version v3.0.0-beta.5](https://github.com/convert-units/convert-units/tree/v3.0.0-beta.5) assuming that's the version you've chosen.

**Comment by neodescis** [comment]
If, like me, you're still trying to use version 2.3.4 and you're getting "global is not defined" errors, you can fix this by forcing 2.3.4 to use later versions of the lodash functions. Here's what I added to my package.json:

```
"overrides": {
    "convert-units": {
        "lodash.foreach": "^4.5.0",
        "lodash.keys": "^4.2.0"
    }
}
```

**Comment by faribauc** [comment]
@Taar Here's how I got it to work with the beta version. A simple typescript Angular service that wraps this package. Can easily be injected in a component for ease of use.

```
import { Injectable } from '@angular/core';
import configureMeasurements from 'convert-units';
import allMeasures from 'convert-units/lib/esm/definitions';

@Injectable({
  providedIn: 'root',
})
export class UnitConversionService {
  _convert;

  constructor() {
    this._convert = configureMeasurements(allMeasures);
  }

  convert(value: number) {
    return this._convert(value);
  }

}
```

Them, from inside the component, you can just call if like so:
```
const newValue = this.convertService.convert(value).from(fromUnit).to(toUnit);
```


Notice now I import `allMeasures`. Contrary to what the doc says, I had to import a specific version of the definitions.
